src/sage/rings/function_field/drinfeld_modules/drinfeld_module.py

```diff
@@ -1532,9 +1696,96 @@ def velu(self, isog):
         if isog == 0:
             raise e
         quo, rem = (isog * self.gen()).right_quo_rem(isog)
-        char_deg = self.characteristic().degree()
-        if not char_deg.divides(isog.valuation()) \
-                or rem != 0:
-            raise e
-        else:
+        if rem.is_zero() and quo[0] == self.gen()[0]:
             return self.category().object(quo)
+        else:
+            raise e
+
+    def hom(self, x, codomain=None):
+        r"""
+        Return the homomorphism defined by ``x`` having this Drinfeld
+        module as domain.
+
+        We recall that a homomorphism `f : \phi \to \psi` between
+        two Drinfeld modules is defined by a Ore polynomial `u`,
+        which is subject to the relation `phi_T u = u \psi_T`.
+
+        INPUT:
+
+        - ``x`` -- an element of the ring of functions, or a
+          Ore polynomial
+
+        - ``codomain`` -- a Drinfeld module or ``None`` (default:
+          ``None``)
+
+        EXAMPLES::
+
+            sage: Fq = GF(5)
+            sage: A.<T> = Fq[]
+            sage: K.<z> = Fq.extension(3)
+            sage: phi = DrinfeldModule(A, [z, 0, 1, z])
+            sage: phi
+            Drinfeld module defined by T |--> z*t^3 + t^2 + z
+
+        An important class of endomorphisms of a Drinfeld module
+        `\phi` is given by scalar multiplications, that are endomorphisms
+        corresponding to the Ore polynomials `\phi_a` with `a` in the function
+        ring `A`. We construct them as follows::
+
+            sage: phi.hom(T)
+            Endomorphism of Drinfeld module defined by T |--> z*t^3 + t^2 + z
+              Defn: z*t^3 + t^2 + z
+
```

Comment:
I don't dislike the preview but if you insist, I can change this.
What's you preference between your two proposals?

Reply:
I think the preview is meant to look like a real shell, so this would not be very realistic (hence my comment). Between the two options, it's a matter of personal preference. I would lean towards `::` but you may choose either, and I'm fine with either.